main.py

```diff
@@ -234,15 +134,14 @@ def gather_summaries(dt, ch, lg, lm, ck, ct, _seldocs):
 with st.expander("Program Inventory"):
   inventory
 
-if f"{ch}-{dt}-{lm}-{lg}.json" in os.listdir("./summaries"):
+if f"{dt}-{ch}-{lm}-{lg}.json" in os.listdir(f"./{OUTPUT_FOLDER_NAME}"):
     print("FOUND FILE")
-    summaries = open(f"./summaries/{ch}-{dt}-{lm}-{lg}.json", "r")
+    summaries = open(f"./{OUTPUT_FOLDER_NAME}/{dt}-{ch}-{lm}-{lg}.json", "r")
     summaries_json = json.loads(summaries.read())
     draw_summaries(summaries_json)
 else:
-
-  seldocs = select_docs(dt, ch, lg, lm, ck, ct)
+  seldocs = select_docs(dt, ch, lg, lm, ck, ct, inventory)
```

Comment:
Did you test these changes? I am afraid the non-hashable (if it is one) parameter (i.e., `inventory`) passed might mess up with the caching system.